### PR TITLE
Added TeamName, Ground Items, Tactic, etc

### PR DESCRIPTION
**Pull request by Jawshoeuh**
Added several functions related to bulk items, the kecleon shop,  using items in ground mode/spinda's cafe, and team name. Also edited the team_member_table for the explorer maze monsters and added a few move related ram addresses.

I've also edited the ground_monster and team_member attack, special attack, defense, and special defense stats to be in line with changes made to them in the dungeon_mode struct for monsters because I found functions that pass the stats address as an argument.

### CODE REVIEW

File: symbols/arm9.yml [hunk 1]
```diff
@@ -1450,6 +1450,32 @@ arm9:
...
+        r0: pointer to item to initialize
+        r1: pointer to bulk_item
+    - name: ItemToBulkItem
+      address:
+        NA: 0x200D0A0
+      description: |-
+        Convert an item into an equivalent bulk_item.
+        
+        r0: pointer to item to initialize
+        r1: pointer to bulk_item
```
Comment (UsernameFodder) [inline symbols/arm9.yml:1478]: One of these doesn't match the parameter order defined in the headers. Not sure which one is wrong.

File: headers/functions/arm9.h [hunk 2]
```diff
@@ -179,8 +182,8 @@ enum move_id GetItemMoveId(enum item_id item_id);
 bool TestItemAiFlag(enum item_id item_id, int bit_id);
 bool IsItemInTimeDarkness(enum item_id item_id);
 bool IsItemValidVeneer(enum item_id item_id);
-void SetGold(int val);
-int GetGold(void);
+void SetActiveInventory(int inventory);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:185]: Does `enum team_id` fit here for the parameter?

File: symbols/arm9.yml [hunk 3]
```diff
@@ -2240,6 +2267,13 @@ arm9:
         
         r0: pointer to an owned_item
         return: bool
+    - name: GetMoneyStored
```
Comment (UsernameFodder) [inline symbols/arm9.yml:2270]: Can you add a function prototype for this?

File: symbols/arm9.yml [hunk 4]
```diff
@@ -2249,18 +2283,48 @@ arm9:
         Sets the amount of money the player has stored in the Duskull Bank, clamping the value to the range [0, MAX_MONEY_STORED].
         
         r0: new value
-    - name: GetKecleonItems1
+    - name: AddMoneyStored
```
Comment (UsernameFodder) [inline symbols/arm9.yml:2286]: Can you add a function prototype for this?

File: symbols/arm9.yml [hunk 5]
```diff
@@ -2344,33 +2408,135 @@ arm9:
...
+        r0: ground monster pointer
+        r1: [output] pointer to attempted hp boost, if not NULL
+        return: actual hp boost
+    - name: ApplyGinsengToGroundMonster
       address:
         EU: 0x20116A0
         NA: 0x20115F8
         JP: 0x20115C8
       description: |-
         Note: unverified, ported from Irdkwia's notes
```
Comment (UsernameFodder) [inline symbols/arm9.yml:2457]: Is this still unverified?

File: headers/functions/arm9.h [hunk 6]
```diff
@@ -234,10 +239,33 @@ bool IsExclusiveItemForMonster(struct item* item, enum monster_id monster_id, en
...
+int ApplyIronBoostToGroundMonster(struct ground_monster* ground_monster,
+                                  int* attempted_defense_boost_out);
+int ApplyZincBoostToGroundMonster(struct ground_monster* ground_monster,
+                                  int* attempted_sp_defense_boost_out);
+int ApplyNectarBoostToGroundMonster(struct ground_monster* ground_monster,
+                                    int* attempted_iq_boost_out);
+bool IsMonsterAffectedByGravelyrockGroundMode(struct ground_monster* ground_monster);
+int ApplyGravelyrockBoostToGroundMonster(struct ground_monster* ground_monster,
+                                         int* attempted_iq_boost_out);
+void ApplyGummiBoostsGroundMode(uint16_t* monster_id, uint16_t* monster_iq,
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:265]: Does `struct monster_id_16` apply for the first parameter?

File: headers/functions/arm9.h [hunk 7]
```diff
@@ -422,6 +450,7 @@ int StoiTagVeneer(const char* s);
 void InitPreprocessorArgs(struct preprocessor_args* args);
 char* SetStringAccuracy(char* s, int param_2);
 char* SetStringPower(char* s, int param_2);
+char* GetCurrentTeamNameString(char* buffer);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:453]: The description says there's a second parameter. Is that missing here?

File: symbols/arm9.yml [hunk 8]
```diff
@@ -4141,10 +4318,21 @@ arm9:
...
+        A special version of strcpy for handling names. Appears to use character 0x7E as some kind of
+        formatting character in NA?
+        
+        r0: dst
+        r1: src
+    - name: SpecialNStrcpy
+      address:
+        NA: 0x2025314
+      description: |-
+        A special version of strcpy for handling names. Appears to use character 0x7E as some kind of
```
Comment (UsernameFodder) [inline symbols/arm9.yml:4330]: Should this be:

```suggestion
        A special version of strncpy for handling names. Appears to use character 0x7E as some kind of
```

?

File: headers/functions/arm9.h [hunk 9]
```diff
@@ -431,13 +460,16 @@ void StrncpySimpleNoPad(char* dest, const char* src, uint32_t n);
 int StrncmpSimple(const char* s1, const char* s2, uint32_t n);
 void StrncpySimpleNoPadSafe(char* dest, const char* src, uint32_t n);
 void SpecialStrcpy(char* dest, const char* src);
+void SpecialNStrcpy(char* dest, const char* src, uint32_t n);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:463]: Can we maybe name these better? How about `StrcpyName` and `StrncpyName`?

File: symbols/arm9.yml [hunk 10]
```diff
@@ -5887,17 +6100,17 @@ arm9:
...
+        r0: [output] buffer
         r1: ability ID
     - name: GetAbilityDescStringId
       address:
         EU: 0x2050C74
         NA: 0x205093C
         JP: 0x2050C88
       description: |-
-        Note: unverified, ported from Irdkwia's notes
+        Gets the ability descripting string ID for the corresponding ability.
```
Comment (UsernameFodder) [inline symbols/arm9.yml:6113]:
```suggestion
        Gets the ability description string ID for the corresponding ability.
```

File: headers/functions/arm9.h [hunk 11]
```diff
@@ -611,12 +644,13 @@ void CopyLogFrom(undefined* read_info);
 void GetAbilityString(undefined* buf, enum ability_id ability_id);
 int GetAbilityDescStringId(enum ability_id ability_id);
 int GetTypeStringId(enum type_id type_id);
+int GetConversion2ConvertToType(enum type_id attack_type_id);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:647]: Should the return type be `enum type_id` as well?

File: symbols/arm9.yml [hunk 12]
```diff
@@ -7393,6 +7638,38 @@ arm9:
...
+    - name: GetUnlockedTactics
+      address:
+        NA: 0x2058C4C
+      description: |-
+        Returns an array with all the enabled tactics. TACTIC_NONE is used to fill the empty/unused entries
+        in the array.
+        
+        r0: [output] Array of tactic_ids that are enabled
+        r1: Monster level
+    - name: GetUnlockedTacticFlags
```
Comment (UsernameFodder) [inline symbols/arm9.yml:7665]: Can you add a function prototype for this?

File: symbols/arm9.yml [hunk 13]
```diff
@@ -7548,6 +7850,16 @@ arm9:
...
+    - name: CopyTacticString
+      address:
+        NA: 0x205A430
+      description: |-
+        Gets the string corresponding to a given message ID and copies it to the buffer specified in r0.
+        
+        This function won't write more than 64 bytes.
+        
+        r0: [output] buffer
+        r1: tactic_id
```
Comment (UsernameFodder) [inline symbols/arm9.yml:7862]: The headers say this returns `bool`. Is that missing here?

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/dungeon_mode/dungeon.h [hunk 14]
```diff
@@ -1464,8 +1464,8 @@ struct dungeon {
     uint16_t n_normal_item_spawns;
     bool gravity; // 0x12AFC: Gravity is in effect
     undefined field_0x12afd;
-    undefined field_0x12afe;
-    undefined field_0x12aff;
+    // 0x12AFE: Victories on one floor counter.
```
[the remark this answers is not in the document]
Comment (Jawshoeuh) [reply headers/types/dungeon_mode/dungeon.h:1467]: There's a stat that keeps track of the most number of 'victories' on one floor. I believe it's basically the number of monsters the team has knocked out on each floor. I'm not sure where this stat is used in game, but it's added to every time a monster is knocked out and compared against the current saved 'victories' if it's higher it gets saved as the new highest victory count. Do you want me to just add this clarification and leave the name alone?

File: headers/functions/arm9.h (hunk 9 above)
Comment (Jawshoeuh) [reply headers/functions/arm9.h:463]: Yes, those are better names. Will make appropriate changes.

File: headers/functions/arm9.h (hunk 2 above)
Comment (Jawshoeuh) [reply headers/functions/arm9.h:185]: Yes, that is a better fit.

File: symbols/arm9.yml [hunk 15]
```diff
@@ -2240,6 +2267,13 @@ arm9:
         
         r0: pointer to an owned_item
         return: bool
+    - name: GetMoneyStored
+      address:
+        NA: 0x201070C
+      description: |-
+        Gets the amount of money the player has stored in the Duskull Bank.
+        
+        r0: new value
```
Comment (UsernameFodder) [inline symbols/arm9.yml:2276]: This seems wrong. I think this should be something like `return: money stored`?

File: symbols/arm9.yml (hunk 4 above)
Comment (UsernameFodder) [reply symbols/arm9.yml:2286]: This prototype still seems to be missing.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
Just a few more things. See the remaining open comments.

File: headers/types/dungeon_mode/dungeon.h (hunk 14 above)
[the remark this answers is not in the document]
Comment (UsernameFodder) [reply headers/types/dungeon_mode/dungeon.h:1467]: Oh I see. Yeah we can just leave this alone.

File: symbols/arm9.yml [hunk 16]
```diff
@@ -7286,6 +7510,27 @@ arm9:
...
+        Removes the active monsters on the Special Episode Team from the team member table.
+        
+        No params.
+    - name: RemoveActiveMembersFromRescueTeam
+      address:
+        NA: 0x20561C0
+      description: |-
+        Removes the active monsters on the Rescue Team from the team member table.
+        
+        No params.
```
[the remark this answers is not in the document]
Comment (UsernameFodder) [reply symbols/arm9.yml:7533]: This prototype still seems to be missing.

**Review by UsernameFodder — APPROVED** [review]
I filled in the last couple. Looks good now, thanks!

**Pull request merged**